test/infer/test_gradient.py

```diff
@@ -366,3 +366,90 @@ def actual_loss_fn(params_raw):
 
             assert_equal(actual_loss, expected_loss, prec=1e-5)
             assert_equal(actual_grads, expected_grads, prec=1e-5)
+
+
+def kl_model_8_z1z2(params):
```

Comment:
can we explicitly compare the corresponding ELBO with mean field guide directly to something like `kl_div(Cat, Cat) + kl_div(Norm, Norm)` in the test?